src/App.tsx

```diff
@@ -154,12 +155,43 @@ const App: React.FC = () => {
     //Checking for flexible update in play-store
     Util.startFlexibleUpdate();
 
-    //Checking for Notification permissions
-    Util.checkNotificationPermissions();
-
     //Listen to network change
     Util.listenToNetwork();
 
+    Util.checkNotificationPermissionsAndType(async (type, rewardProfileUid) => {
+      const currentStudent = Util.getCurrentStudent();
+      if (!rewardProfileUid) {
+        console.error("Reward profile UID is undefined");
+        return;
+      }
+      const students =
+        await ServiceConfig.getI().apiHandler.getParentStudentProfiles();
+      console.log(
+        "🚀 ~ file: DisplayStudents.tsx:13 ~ getStudents ~ students:",
+        students
+      );
+      const docIds = students.map((student) => student.docId);
+      if (type) {
+        if (type === "reward") {
```

Comment:
check this condition at start of the function and include the code inside if condition so that we can add other notification callbacks